### DISCUSSION TITLE
Definições de Arquitetura.

### DISCUSSION THREAD

**Discussion by n1arthn1** [balta-io/7196#42] [Perguntas e Respostas] [answered]
Olá, 
Poderia me informar quais design patterns estamos ultilizando durante o curso.
Criacional:
Comportamental:
Arquitetural:

Obrigado.

**Comment by andrebaltieri** [comment, marked as answer]
@n1arthn1 todos... não todos que existem, mas tem criacionais como factory e builder utilizados nos notifications e no flunt. Tem adapter, decorator e facade usados nos handlers e comportamentais como commands.

**Reply by n1arthn1** [reply]
Muito Obrigado, eu havia estudado e verificado que foram utilizados vários padrões, mas não tinha certeza... mas agora concluo que é factível e necessário o uso de vários padrões.

**Reply by n1arthn1** [reply]
Ao recebermos um ViewModel via argumento, validar, e persistir no banco, podemos dizer que estamos utilizando Singleton.?!?

**Reply by andrebaltieri** [reply]
Bom dia @n1arthn1 como vai? Não, o singleton é outra coisa... vale a pena conferir o curso Dominando Injeção de dependência, falo mais sobre ele lá...
